Makefile

```diff
@@ -61,13 +66,34 @@ golangci-lint:
 
 .PHONY: gen-prom
 gen-prom: clean
+	rm -rf ${PROM_OUT_PATH}
 	rm -rf ${PROM_MANIFESTS_PATH}
 	mkdir -p ${PROM_MANIFESTS_PATH}
 	mkdir -p ${PROM_OUT_PATH}
-	cp manifests/jsonnet/kube-prometheus.jsonnet ${PROM_OUT_PATH}
-	hack/gen-prom.sh ${PROM_OUT_PATH} ${KUBE_PROM_VER} ${KUBE_PROM_CFG_FILE}
+	cp manifests/jsonnet/prometheus/prometheus.jsonnet ${PROM_OUT_PATH}/kube-prometheus.jsonnet
+	hack/gen-prom.sh ${PROM_OUT_PATH} ${KUBE_PROM_VER} kube-prometheus.jsonnet
 	cp -r ${PROM_OUT_PATH}/manifests/* ${PROM_MANIFESTS_PATH}
 
+.PHONY: gen-prom-thanos
+gen-prom-thanos:
+	rm -rf ${PROM_THANOS_OUT_PATH}
+	rm -rf ${PROM_THANOS_MANIFESTS_PATH}
+	mkdir -p ${PROM_THANOS_MANIFESTS_PATH}
+	mkdir -p ${PROM_THANOS_OUT_PATH}
+	cp manifests/jsonnet/prometheus/thanos.jsonnet ${PROM_THANOS_OUT_PATH}/kube-prometheus.jsonnet
+	hack/gen-prom.sh ${PROM_THANOS_OUT_PATH} ${KUBE_PROM_VER} kube-prometheus.jsonnet
+	cp -r ${PROM_THANOS_OUT_PATH}/manifests/* ${PROM_THANOS_MANIFESTS_PATH}
+
+.PHONY: gen-thanos
+gen-thanos:
```

Comment:
Suggest moving component gen out, in this root makefile we only need to include that specific components files

Reply:
will do that in separate PR, same to prometheus, let's track this by issue